src/geo/map/cartodb-layer.js

```diff
@@ -31,6 +45,17 @@ var CartoDBLayer = LayerModelBase.extend({
     return this.get('visible');
   },
 
+  getGeometryType: function () {
```

Comment:
and I'd remove this method, basically guess the geometry type from the style set. Nothing avoids me to set a style with marker-width rendering polygons (we actually do it in bubble maps).